src/features/accuse/useAccusedWallets.ts

```diff
@@ -0,0 +1,58 @@
+import { Signature, Wallet } from 'ethers';
+import { defaultAbiCoder, splitSignature } from 'ethers/lib/utils';
+import { isBytes32 } from 'lib/utils/helpers';
+import { useEffect, useState } from 'react';
+
+/**
+ * A hook that creates signatures for each private key provided including sarcophagusId and
+ * paymentAddress in the message. Updates automatically when a private key is changed.
+ * @param sarcophagusId the id of the sarcophagus
+ * @param archaeologistPrivateKeys the private keys of the archaeologists
+ * @param paymentAddress the address to pay the sarcophagus to
+ * @returns an array of signatures
+ */
+export function useAccusedWallets(
+  sarcophagusId: string,
+  archaeologistPrivateKeys: string[],
+  paymentAddress: string | undefined
+): { signatures: Signature[]; publicKeys: string[] } {
+  const [signatures, setSignatures] = useState<Signature[]>([]);
+  const [publicKeys, setPublicKeys] = useState<string[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const newSignatures: Signature[] = [];
+        const newPublicKeys: string[] = [];
+
+        for (let i = 0; i < archaeologistPrivateKeys.length; i++) {
+          const untrimmedPrivateKey = archaeologistPrivateKeys[i];
+          const privateKey = untrimmedPrivateKey.trim();
+
+          // Only sign the messages if the values are valid
+          if (!isBytes32(privateKey) || !isBytes32(sarcophagusId)) {
+            return;
+          }
```

Comment:
Does the user see any validation messages if one of their keys is malformed? I.e. what happens if keys 1 and 3 are valid, but 2 is invalid? Would it still allow me to click "accuse"?

Reply:
The accuse button is enabled only if the estimate gas part is successful, which is triggered anytime a field is changed. So as long as the inputs would fail on the contract, the accuse button will be disabled. 

There are no validation messages, however.